Default package causes problems

```
java.lang.RuntimeException: Failed to backport class: Example
    at net.orfjackal.retrolambda.Transformers.transform(Transformers.java:128)
    at net.orfjackal.retrolambda.Transformers.transform(Transformers.java:107)
    at net.orfjackal.retrolambda.Transformers.backportClass(Transformers.java:47)
    at net.orfjackal.retrolambda.Retrolambda.run(Retrolambda.java:77)
    at net.orfjackal.retrolambda.Main.main(Main.java:26)
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: -1
    at java.lang.String.substring(String.java:1955)
    at net.orfjackal.retrolambda.lambdas.BackportLambdaInvocations.isNonOwnedMethodVisible(BackportLambdaInvocations.java:107)
    at net.orfjackal.retrolambda.lambdas.BackportLambdaInvocations.getLambdaAccessMethod(BackportLambdaInvocations.java:66)
    at net.orfjackal.retrolambda.lambdas.BackportLambdaInvocations$InvokeDynamicInsnConverter.backportLambda(BackportLambdaInvocations.java:169)
    at net.orfjackal.retrolambda.lambdas.BackportLambdaInvocations$InvokeDynamicInsnConverter.visitInvokeDynamicInsn(BackportLambdaInvocations.java:160)
    at net.orfjackal.retrolambda.asm.ClassReader.readCode(ClassReader.java:1452)
    at net.orfjackal.retrolambda.asm.ClassReader.readMethod(ClassReader.java:1017)
    at net.orfjackal.retrolambda.asm.ClassReader.accept(ClassReader.java:693)
    at net.orfjackal.retrolambda.asm.ClassReader.accept(ClassReader.java:506)
    at net.orfjackal.retrolambda.Transformers.lambda$transform$5(Transformers.java:107)
    at net.orfjackal.retrolambda.Transformers$$Lambda$8/1530388690.accept(Unknown Source)
    at net.orfjackal.retrolambda.Transformers.transform(Transformers.java:124)
    ... 4 more
```

Pretty sure this is my fault! We should be checking for `-1` from `indexOf` before calling `substring`. Unlikely to be hit in production because no one should be using the default package! I found it while working on content for a presentation.

I'll send a fix later this week or next if someone doesn't beat me to it.

--- a/end-to-end-tests/src/test/java/DefaultPackageTest.java
+++ b/end-to-end-tests/src/test/java/DefaultPackageTest.java
@@ -1,0 +1,19 @@
+// Copyright © 2013-2016 Esko Luontola and other Retrolambda contributors
+// This software is released under the Apache License 2.0.
+// The license text is at http://www.apache.org/licenses/LICENSE-2.0
+
+import org.junit.Test;
+
+public final class DefaultPackageTest {
+    @Test
+    public void method_reference_to_sibling_class() {
+        SiblingClass sibling = new SiblingClass();
+        Runnable lambda = sibling::method;
+        lambda.run();
+    }
+}
+
+class SiblingClass {
+    void method() {
+    }
+}

--- a/retrolambda/src/main/java/net/orfjackal/retrolambda/lambdas/BackportLambdaInvocations.java
+++ b/retrolambda/src/main/java/net/orfjackal/retrolambda/lambdas/BackportLambdaInvocations.java
@@ -104,8 +104,12 @@ public class BackportLambdaInvocations extends ClassVisitor {
     }
 
     private boolean isNonOwnedMethodVisible(Handle implMethod) {
-        String classPackage = className.substring(0, className.lastIndexOf('/'));
-        String implPackage = implMethod.getOwner().substring(0, implMethod.getOwner().lastIndexOf('/'));
+        int classNameLastSlash = className.lastIndexOf('/');
+        String classPackage = classNameLastSlash == -1 ? "" : className.substring(0, classNameLastSlash);
+
+        int implMethodLastSlash = implMethod.getOwner().lastIndexOf('/');
+        String implPackage = implMethodLastSlash == -1 ? "" : implMethod.getOwner().substring(0, implMethodLastSlash);
+
         if (classPackage.equals(implPackage)) {
             return true; // All method visibilities in the same package will be visible.
         }

